### ISSUE TITLE
Modifying draft by spreading property in draft state not working in 1.0.0

### ISSUE THREAD

**Issue by pbomb** [mweststrate/immer#89]
I just upgraded from 0.8.5 to 1.0.0 and am seeing a regression. I'm updating an object within the `produce` function by overwriting a property on that object which is itself an object. When I spread the nested object in the draft object at that same property, immer does not seem to make the changes. If the property didn't exist before, it continues to not exist. If the property does exist, the changes aren't reflected in the resulting object. This worked properly in 0.8.5.

I've set up [this sandbox](https://codesandbox.io/s/4x4v9j1x94) to reproduce the issue. Here's a small snippet that doesn't work as expected.
```js
const state = {
  users: {
    existingID: {
      loading: false,
      value: 'val'
    }
  }
}
const nextState = produce(state, draft => {
  draft.users.existingID = {
    ...draft.users.existingID,
    loading: true
  }
})
console.log(nextState.users.existingID.loading) // prints "false" instead of "true"
```

**Comment by mweststrate** [comment]
Thanks for reporting, will investigate!

**Comment by mweststrate** [comment]
Should be fixed in 1.0.1. Same issue as found and fixed by @iruca3  through #86. Just released. Feel free to reopen if it isn't

**Comment by pbomb** [comment]
Thanks for the quick turnaround. I was able to verify that the new version does resolve this issue.
